### PR TITLE
Initial CSS module implementation

### PR DESCRIPTION
**Pull request by bitpshr**
After talking with @Tomdye last week, I worked up this PR to demonstrate how CSS modules would be used to accomplish styling for the Monster Cards app. Development felt slow and roundabout, though that could've been due to the fact that I was converting existing styles rather than implementing new styles.

Some notes:
* For now, I didn't update the sprite sheets or `global.styl`.
* New CSS module typings had to be generated every time I changed a CSS module via `dojo css typings`,
* When using [class composition](https://github.com/css-modules/css-modules#composition), a string of space-separated CSS classes is exported by webpack, causing some rather ugly template strings.

Overall, using CSS modules for a one-off application like this seemed to cause more problems than it solved, as Monster Cards already had Stylus written that had no global conflicts. I can kind of see the benefit for a set of widgets shared across many applications (even though a single, top-level unique selector on a shared widget would be sufficient 99% of the time), but for a small app like this, it seems like an unnecessary burden to impose on developers.

Refs #60

### CODE REVIEW

**Comment by agubler** [comment]
@bitpshr the generated css module interfaces are causing CI to fail for not ending with a newline.

File: monster-cards/src/widgets/card/createCardSummary.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { Widget, DNode, WidgetState } from 'dojo-interfaces/widgetBases';
 import createWidgetBase from 'dojo-widgets/bases/createWidgetBase';
 import d from 'dojo-widgets/util/d';
 import createCard from './createCard';
+import * as css from './card-summary.module.styl';
+
+console.log(css);
```
Comment (agubler) [inline monster-cards/src/widgets/card/createCardSummary.ts]: console.log

File: monster-cards/src/widgets/about/createAboutPage.ts [hunk 2]
```diff
@@ -13,7 +14,7 @@ const createAboutPage = createWidgetBase
 			childNodeRenderers: [
 				function(this: AboutPage): DNode[] {
 					const heading = d('h1', { innerHTML: 'About' });
-					const jumbotron = d('div.jumbotron', {}, [ heading ]);
+					const jumbotron = d(`div.${ css.jumbotron.replace(/\s/g, '.') }`, {}, [ heading ]);
```
Comment (agubler) [inline monster-cards/src/widgets/about/createAboutPage.ts]: I thought we were going to write a reducer to return the correct class structure for the options?

**Comment by bitpshr** [comment]
@Tomdye @agubler I cleaned up this PR a bit since we may land it:

- Used a reducer for composed CSS class names.
- Modularized everything but the sprite sheets and the reset. We could move these imports into `genera.module.styl` so `main.styl` has no content.
- Removed logs and addressed other PR feedback

Also note that this work depends on dojo/cli-css-typings#5 and dojo/cli-build#20 and both are ready for another review.

**Comment by agubler** [comment]
@bitpshr This is quite a bit out of date and possibly will be superseded once the proposals that @Tomdye has been working have been implemented, so am closing.

**Pull request closed without merging**